### ISSUE TITLE
Wagtail should not change month name translations

### ISSUE THREAD

**Issue by bmihelac** [wagtail/wagtail#6292]
### Issue Summary

Wagtail translations overrides month name translations (at least for Slovenian language) which changes how dates are formatted.

### Steps to Reproduce

With wagtail installed:

```python
>>> from django.utils.translation import activate                                                                                                
>>> activate("sl")                                                                                                                               
>>> from django.utils import formats 
...                                                                                                                                              
>>> from datetime import date                                                                                                                    
>>> formats.date_format(date.today())                                                                                                            
'5. Avgust 2020'
```

It should be (and without wagtail installed it is) `5. avgust 2020`.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no)

yes

### Technical details

* Python version: Run `python --version`.

Python 3.7.1

Django version: Look in your requirements.txt, or run `pip show django | grep Version`.

Version: 2.2.14

* Wagtail version: Look at the bottom of the Settings menu in the Wagtail admin, or run `pip show wagtail | grep Version:`.

Version: 2.9.2

**Comment by bmihelac** [comment]
This is visible in translation files:

Django

https://github.com/django/django/blob/master/django/conf/locale/sl/LC_MESSAGES/django.po#L919-L953

Wagtail

https://github.com/wagtail/wagtail/blob/b6455ecfb9d91b4a8f36f4f579dc969b1f705eba/wagtail/admin/locale/sl/LC_MESSAGES/django.po#L197-L231

**Comment by gasman** [comment]
Thanks for the report @bmihelac! Have submitted a proposed fix at #6301 now.

**Comment by thibaudcolas** [comment]
I’ve merged #6301, this should be fixed in the upcoming 2.11 release.

Note that we will need to update Wagtail’s .po files in order for this to really be fixed, which only happens as part of the release process, so the problem will still be present on `master` between now and the 2.11 stable release.
